x-pack/legacy/plugins/apm/public/components/shared/Summary/DurationSummaryItem.tsx

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiToolTip, EuiText } from '@elastic/eui';
+import { PercentOfParent } from '../../app/TransactionDetails/WaterfallWithSummmary/PercentOfParent';
+import { asTime } from '../../../utils/formatters';
+
+interface Props {
+  duration: number;
+  totalDuration: number | undefined;
+  parentType: 'trace' | 'transaction';
+}
+
+const DurationSummaryItem = ({
+  duration,
+  totalDuration,
+  parentType
+}: Props) => {
+  const calculatedTotalDuration =
+    totalDuration === undefined ? duration : totalDuration;
```

Comment:
This is a good use-case for the null coalesce operator you ranted about :p

Reply:
I really like that operator! I'm just empathetic towards the people that have to write the parsers 😬